File: BIND_core/src/+fr/+lescot/+bind/+data/MetaEvent.m
%{
Class:

<MetaBase> extension with the specific properties of the events (none at
the moment).

%}
classdef MetaEvent < fr.lescot.bind.data.MetaBase
    
    methods
        
        %{
        Function:
        Instanciates a new MetaEvent object, and set the frameworkVariables
        to the correct values.
        
        Returns:
        A MetaBase instance
        %}
        function out = MetaEvent()
            out@fr.lescot.bind.data.MetaBase;
            timecode = fr.lescot.bind.data.MetaEventVariable();
            timecode.setName('timecode');
            timecode.setType(fr.lescot.bind.data.MetaDataVariable.TYPE_REAL);
            out.setFrameworkVariables({timecode});
        end
        
        %{
        Function:
        Returns a hash of the object, which is an identifier string which
        is equal between two objects only if the two objects are equal
        (equality being here an esuality of value, not of reference).
        
        Arguments:
        this - The object on which the function is called, optionnal.
        
        Returns:
        A string   
        %}     
        function out = hash(this)
            baseClassHash = hash@fr.lescot.bind.data.MetaBase(this);
            out = [baseClassHash '|' 'EVENT'];
        end
        
    end
    
end

File: pynd/pynd/data/meta_event.py
from pynd.data import MetaBase, MetaEventVariable, MetaVariableBaseType


class MetaEvent(MetaBase):

    def __init__(self):
        super().__init__()
        timecode = MetaEventVariable()
        timecode.set_name('timecode')
        timecode.set_type(MetaVariableBaseType.REAL)
        self._set_framework_variables([timecode])

    def hash(self) -> str:
        return super(MetaEvent, self).hash() + '|' + 'EVENT'
